internal/rpcserver/rpcserverhandlers_test.go

```diff
@@ -4319,6 +4319,64 @@ func TestHandleGetWork(t *testing.T) {
 	}})
 }
 
+func TestHandleSetGenerate(t *testing.T) {
+	t.Parallel()
+
+	procLimit := 2
+	miningaddr, err := dcrutil.DecodeAddress("DsRM6qwzT3r85evKvDBJBviTgYcaLKL4ipD", defaultChainParams)
+	if err != nil {
+		t.Fatalf("[DecodeAddress] unexpected error: %v", err)
+	}
+
+	testRPCServerHandler(t, []rpcTest{{
+		name:    "handleSetGenerate: no payment addresses",
+		handler: handleSetGenerate,
+		cmd: &types.SetGenerateCmd{
+			Generate:     true,
+			GenProcLimit: &procLimit,
+		},
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInternal.Code,
+	}, {
+		name:    "handleSetGenerate: ok",
+		handler: handleSetGenerate,
+		cmd: &types.SetGenerateCmd{
+			Generate:     true,
+			GenProcLimit: &procLimit,
+		},
+		mockMiningState: func() *testMiningState {
+			ms := defaultMockMiningState()
+			ms.miningAddrs = []dcrutil.Address{miningaddr}
+			return ms
+		}(),
+	}, {
+		name:    "handleSetGenerate: ok, generate=false",
+		handler: handleSetGenerate,
+		cmd: &types.SetGenerateCmd{
+			Generate: false,
+		},
+	}})
+}
```

Comment:
Looks like one code path is being missed for this one, where `*GenProcLimit` is set to `0`, causing it to hit [this code path](https://github.com/decred/dcrd/blob/master/internal/rpcserver/rpcserver.go#L4245) and set `generate` to `false`.